### PR TITLE
(FM-4969) aix auto tests for physical/logical volume and volume_group on aix

### PR DESCRIPTION
**Pull request by phongdly**

### CODE REVIEW

**Comment by phongdly** [comment]
Because of ticket BKR-717, let me know if you like to run the PR, I'll need to setup the boxes first.

File: tests/beaker/lib/lvm_helper.rb [hunk 1]
```diff
@@ -33,6 +33,23 @@ def verify_if_created?(agent, resource_type, resource_name, vg=nil, properties=n
           assert_match(/#{properties}/, result.stdout, 'Unexpected error was detected')
         end
       end
+    when 'aix_physical_volume'
+      on(agent, "lspv #{resource_name}") do |result|
+        assert_match(/Physical volume #{resource_name} is not assigned to/, result.stdout, 'Unexpected error was detected')
+      end
+    when 'aix_volume_group'
+      on(agent, "lsvg") do |result|
+        #assert_match(/#{resource_name}\s+^a-z0-9$\s+#{vg}/, result.stdout, 'Unexpected error was detected')
```
Comment (cowofevil) [inline tests/beaker/lib/lvm_helper.rb]: Remove dead code.

File: tests/beaker/lib/lvm_helper.rb [hunk 2]
```diff
@@ -33,6 +33,23 @@ def verify_if_created?(agent, resource_type, resource_name, vg=nil, properties=n
...
+        assert_match(/#{resource_name}/, result.stdout, 'Unexpected error was detected')
+      end
+    when 'aix_logical_volume'
+      fail_test "Error: missing volume group that the logical volume is associated with" unless vg
```
Comment (cowofevil) [inline tests/beaker/lib/lvm_helper.rb]: The method throws exceptions everywhere else except here. I think you should change this to an exception so it can be caught by the caller.

File: tests/beaker/lib/lvm_helper.rb [hunk 3]
```diff
@@ -33,6 +33,23 @@ def verify_if_created?(agent, resource_type, resource_name, vg=nil, properties=n
           assert_match(/#{properties}/, result.stdout, 'Unexpected error was detected')
         end
       end
+    when 'aix_physical_volume'
```
Comment (cowofevil) [inline tests/beaker/lib/lvm_helper.rb]: These new resource types are not documented in the method doc string.

File: tests/beaker/tests/aix/create_lv_with_param_max_range_aix.rb [hunk 4]
```diff
@@ -0,0 +1,49 @@
+require 'master_manipulator'
+require 'lvm_helper'
+require 'securerandom'
+
+test_name "FM-4969 - C97367 - create logical volume with parameter min 'range' aix"
```
Comment (cowofevil) [inline tests/beaker/tests/aix/create_lv_with_param_max_range_aix.rb]: Incorrect TestRail ID specified and test name does not match. I think you mixed this up with C97372.

File: tests/beaker/tests/aix/create_lv_with_param_min_range_aix.rb [hunk 5]
```diff
@@ -0,0 +1,49 @@
+require 'master_manipulator'
+require 'lvm_helper'
+require 'securerandom'
+
+test_name "FM-4969 - C97372 - create logical volume with parameter min 'range' aix"
```
Comment (cowofevil) [inline tests/beaker/tests/aix/create_lv_with_param_min_range_aix.rb]: Incorrect TestRail ID specified and test name does not match. I think you mixed this up with C97367.

File: tests/beaker/lib/lvm_helper.rb [hunk 6]
```diff
@@ -78,41 +96,47 @@ def is_correct_format?(agent, volume_group, logical_volume, format_type)
...
 #
 # remove_all(agent, '/dev/sdb', 'VolumeGroup_1234', 'LogicalVolume_fa13')
-def remove_all(agent, pv=nil, vg=nil, lv=nil)
-  step 'remove logical volume if any:'
-  if lv
-    if lv.kind_of?(Array)
-      lv.each do |logical_volume|
-        on(agent, "umount /dev/#{vg}/#{logical_volume}", :acceptable_exit_codes => [0,1])
-        on(agent, "lvremove /dev/#{vg}/#{logical_volume} --force")
+def remove_all(agent, pv=nil, vg=nil, lv=nil, aix=nil)
```
Comment (cowofevil) [inline tests/beaker/lib/lvm_helper.rb]: You should have the "aix" parameter be a boolean flag instead of a string parameter. So the default would be ```aix=false```.

File: tests/beaker/tests/aix/create_lv_with_param_type.rb [hunk 7]
```diff
@@ -0,0 +1,49 @@
+require 'master_manipulator'
+require 'lvm_helper'
+require 'securerandom'
+
+test_name "FM-4969 - C97368 - create logical volume with param 'type' aix"
```
Comment (cowofevil) [inline tests/beaker/tests/aix/create_lv_with_param_type.rb]: Test name is misleading. The type is 'jfs' and not 'aix'. I would reword the title to indicate that 'jfs' is AIX specific.

File: tests/beaker/tests/aix/create_lv_with_param_type.rb [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+volume_group {'#{vg}':
+  ensure            => present,
+  physical_volumes  => #{pv}
+}
+->
+logical_volume{'#{lv}':
+  ensure        => present,
+  volume_group  => '#{vg}',
+  initial_size  => '5M',
+  type         => 'jfs',
```
Comment (cowofevil) [inline tests/beaker/tests/aix/create_lv_with_param_type.rb]: Misaligned hash rocket.

File: tests/beaker/tests/aix/create_physical_volume_on_aix.rb [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+  agents.each do |agent|
+    expect_failure('expected to faile due to FM-4911') do
+      on(agent, "/opt/puppetlabs/puppet/bin/puppet agent -t --environment production", :acceptable_exit_codes => [1]) do |result|
+        assert_no_match(/Error:/, result.stderr, 'Unexpected error was detected!')
+      end
+    end
+
+    step "Verify the physical volume  is created on AIX box: #{pv}"
+    # comment out the below verify function due to FM-4911
+    #verify_if_created?(agent, 'aix_physical_volume', pv)
```
Comment (cowofevil) [inline tests/beaker/tests/aix/create_physical_volume_on_aix.rb]: This should be an expect failure. If you made ```verify_if_created?(agent, 'aix_physical_volume', pv)``` throw an exception then you could utilize the Beaker built-in ```expect_failure``` which would allow the test to pass now. In the future when the issue is fixed the test will break indicating it needs an update.

Comment (phongdly) [inline tests/beaker/tests/aix/create_physical_volume_on_aix.rb]: This might not be necessary, expect_failure is used in line #23 so the text will break if the ticket is fixed.

Comment (cowofevil) [inline tests/beaker/tests/aix/create_physical_volume_on_aix.rb]: You should then uncomment the code.

Comment (phongdly) [inline tests/beaker/tests/aix/create_physical_volume_on_aix.rb]: Due to ticket FM-4911, this line must be commented, because there is no physical volume created yet until the ticket is resolved
